src/substrate-lib/components/SubstrateLoadingAndErrorHandlingWrapper/index.tsx

```diff
@@ -0,0 +1,74 @@
+import { useDispatch } from 'react-redux';
+import { toast, ToastContainer } from 'react-toastify';
+
+import { isBridgeLoaded } from '@/common/actions/general.actions';
+import FullLoadingSpinner from '@/components/FullLoadingSpinner';
+import InterlayHelmet from '@/parts/InterlayHelmet';
+import { ActionType, ApiStatus, KeyringStatus, useSubstrateState } from '@/substrate-lib/substrate-context';
+
+interface SubstrateLoadingAndErrorHandlingWrapperProps {
+  children: React.ReactNode;
+}
+
+const SubstrateLoadingAndErrorHandlingWrapper = ({
+  children
+}: SubstrateLoadingAndErrorHandlingWrapperProps): JSX.Element => {
+  const dispatch = useDispatch();
+
+  const { apiStatus, apiError, keyringStatus, keyring, api } = useSubstrateState();
+
+  // TODO: translate `FullLoadingSpinner` texts once confirmed
```

Comment:
Currently the texts for loading statuses are hardcoded without being translated. They should be translated once confirmed.